### PR TITLE
feat: add get_connector_manifest command to retrieve connector metadata

### PR DESCRIPTION
**Pull request by aaronsteers**
# feat: add get_connector_manifest command to retrieve connector metadata

## Summary

This PR adds a new MCP tool `get_connector_manifest` to the connector-builder-mcp project that allows retrieval of connector metadata YAML files from the `connectors.airbyte.com` endpoint. The function accepts a connector name (required) and version (optional, defaults to 'latest') and returns the raw metadata content as a string.

**Key implementation details:**
- Uses the URL pattern: `https://connectors.airbyte.com/metadata/airbyte/{connector_name}/{version}/metadata.yaml`
- Supports version prefix cleaning (removes 'v' if present)
- Includes proper HTTP error handling and timeout (30s)
- Follows existing code patterns for tool registration and HTTP requests
- Returns descriptive error messages for failed requests

**Note**: The function name suggests it returns "manifest" files, but it actually fetches "metadata.yaml" files from the connectors.airbyte.com endpoint structure.

## Review & Testing Checklist for Human
- [ ] **Test with multiple connector types**: Verify the URL pattern works for both source and destination connectors (e.g., `source-github`, `destination-postgres`) 
- [ ] **Test version handling**: Try with specific versions, 'latest', and versions with 'v' prefix to ensure proper handling
- [ ] **Verify returned content utility**: Check that the returned metadata.yaml content contains useful information for connector building workflows
- [ ] **Test error scenarios**: Try with invalid connector names, network issues, and non-existent versions to ensure graceful error handling
- [ ] **Consider function naming**: Evaluate if `get_connector_manifest` is the appropriate name given it returns metadata.yaml files, not traditional manifest.yaml files

**Recommended test plan:**
1. Test the new tool via MCP client with known connectors like `source-faker`, `destination-postgres`
2. Verify content format and usefulness for connector development workflows
3. Test edge cases: invalid names, network failures, malformed versions

---

### Diagram

```mermaid
%%{ init : { "theme" : "default" }}%%
graph TD
    Server["connector_builder_mcp/server.py"]:::context
    ConnectorBuilder["connector_builder_mcp/_connector_builder.py"]:::major-edit
    FastMCP["FastMCP App"]:::context
    ExternalAPI["connectors.airbyte.com"]:::context
    
    Server --> |registers tools| FastMCP
    ConnectorBuilder --> |register_connector_builder_tools| FastMCP
    ConnectorBuilder --> |HTTP GET requests| ExternalAPI
    
    subgraph "New Function"
        GetManifest["get_connector_manifest()"]:::major-edit
    end
    
    ConnectorBuilder --> GetManifest
    GetManifest --> |"GET /metadata/airbyte/{name}/{version}/metadata.yaml"| ExternalAPI
    
    subgraph Legend
        L1["Major Edit"]:::major-edit
        L2["Minor Edit"]:::minor-edit  
        L3["Context/No Edit"]:::context
    end

classDef major-edit fill:#90EE90
classDef minor-edit fill:#87CEEB
classDef context fill:#FFFFFF
```

### Notes

- **URL Pattern Discovery**: During development, I initially used an incorrect URL pattern (`files/metadata/airbyte/...`) and had to correct it to `metadata/airbyte/...` after investigating the actual connectors.airbyte.com bucket structure
- **Limited Testing**: Only tested with `destination-amazon-sqs` during development - broader testing across connector types is needed
- **Function vs File Naming**: There's a semantic mismatch where the function is named `get_connector_manifest` but fetches `metadata.yaml` files rather than traditional `manifest.yaml` files

**Link to Devin run**: https://app.devin.ai/sessions/050f37720b8f485a86ad1fbe51d5e013  
**Requested by**: AJ Steers (@aaronsteers)

### CODE REVIEW

File: connector_builder_mcp/_connector_builder.py [hunk 1]
```diff
@@ -452,6 +452,44 @@ def _get_topic_specific_docs(topic: str) -> str:
...
+        version: Version of the connector manifest to retrieve (defaults to 'latest')
+
+    Returns:
+        Raw YAML content of the connector manifest
+    """
+    logger.info(f"Getting connector manifest for {connector_name} version {version}")
+
+    cleaned_version = version.removeprefix("v")
+
+    manifest_url = f"https://connectors.airbyte.com/metadata/airbyte/{connector_name}/{cleaned_version}/metadata.yaml"
```
Comment (aaronsteers) [inline connector_builder_mcp/_connector_builder.py:480]: This should be getting 'manifest.yaml', not 'metadata.yaml'.
